src/SWP/Bundle/ContentBundle/EventListener/ArticleMetadataSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use SWP\Bundle\ContentBundle\ArticleEvents;
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Component\Bridge\Model\PackageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ArticleMetadataSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ArticleEvents::PRE_CREATE => 'populateMetadata',
+        ];
+    }
+
+    /**
+     * @param ArticleEvent $event
+     */
+    public function populateMetadata(ArticleEvent $event)
+    {
+        $package = $this->getPackage($event);
+        $article = $event->getArticle();
+
+        $article->setMetadata([
```

Comment:
okay, let's assume I do that. I would need to execute an extra query to get a Package object based on that package id(which makes no sense because I already have a Package object available here in this case) and then from Package I extract metadata and set it into Article object. And I assume you mean to get the Package metadata directly so I dont need to store it in Article object but rather get it from Package object based on referenced id, right? Then in my opinion its wrong, because the metadata should be saved directly under Article object as we work on that object internally in SWP and you would mix phpcr and orm. A Package is just a class needed for stuff related with converting content that comes from Superdesk into our internal Article object. 

We could add `$packageId` property in Article class and store a reference to Package id - it could be useful in cases we want to locate Article by Package id if the content from Superdesk requires, for example, to kill a package or do whatever is needed.